mlflow/java/spark/src/main/scala/org/mlflow/spark/autologging/DatasourceAttributeExtractor.scala

```diff
@@ -48,11 +48,17 @@ private[autologging] trait DatasourceAttributeExtractorBase {
     if (deltaInfoOpt.isDefined) {
       deltaInfoOpt
     } else {
+      logger.warn("x" * 80)
+      logger.warn(leafNode.getClass().getCanonicalName())
       leafNode match {
-        case relation: DataSourceV2Relation =>
+        case relation: DataSourceV2Relation => {
+          logger.warn("Matched DataSourceV2Relation")
           getSparkTableInfoFromTable(relation.table)
-        case other =>
+        }
+        case other => {
+          logger.warn("Matched other")
           None
+        }
```

Comment:
For spark 3.0-preview, these lines print out:

```
21/04/19 04:51:56 WARN DatasourceAttributeExtractor$: xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
21/04/19 04:51:56 WARN DatasourceAttributeExtractor$: org.apache.spark.sql.execution.datasources.v2.DataSourceV2Relation
21/04/19 04:51:56 WARN DatasourceAttributeExtractor$: Matched DataSourceV2Relation
```

https://github.com/mlflow/mlflow/pull/4262/checks?check_run_id=2377653863#step:5:542

```
21/04/19 04:50:49 WARN DatasourceAttributeExtractor$: xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
21/04/19 04:50:49 WARN DatasourceAttributeExtractor$: org.apache.spark.sql.execution.datasources.LogicalRelation
21/04/19 04:50:49 WARN DatasourceAttributeExtractor$: Matched other
```

https://github.com/mlflow/mlflow/pull/4262/checks?check_run_id=2377653863#step:5:92